benchmarks/bench_auto_batching.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Benchmark suite for joblib
+#
+# Author: Pierre Glaser
+"""Benchmarking the impact of joblib's task batching strategy"""
+import tempfile
+import time
+
+from joblib import Parallel, delayed
+import numpy as np
+
+from .common import Benchmark, sleep_noop
+
+
+def bench_short_tasks(
+    task_times,
+    input_data_size=0,
+    output_data_size=0,
+    memmap_input=False,
+    parallel_inst=None,
+):
+
+    with tempfile.NamedTemporaryFile() as temp_file:
+        if input_data_size:
+            # Generate some input data with the required size
+            if memmap_input:
+                temp_file.close()
+                input_data = np.memmap(
+                    temp_file.name,
+                    shape=input_data_size,
+                    dtype=np.byte,
+                    mode="w+",
+                )
+                input_data[:] = 1
+            else:
+                input_data = np.ones(input_data_size, dtype=np.byte)
+        else:
+            input_data = None
+
+        parallel_inst(
+            delayed(sleep_noop)(max(t, 0), input_data, output_data_size)
+            for t in task_times
+        )
+
+
+class AutoBatchingSuite(Benchmark):
+    repeat = 1
+    number = 1
+    warmup_time = 0
+
+    # In practice, the input size does not influence the benchmarks very much:
+    # The input is a numpy array. Small numpy arrays are pickled very fast and
+    # do not incur significant overhead. Large numpy arrays can take time to be
+    # pickled, but end up being memmapped, which only incurs a one-time
+    # serialization cost, and thus will stop influence the batching after only
+    # a few batches. Therefore, we fix this parameter for now.
+    param_names = ["input_size", "n_jobs"]
+    params = ([10000, 100000, 1000000], [1, 2, 4])
+    parallel_parameters = dict(
+        verbose=10, backend="loky", pre_dispatch="2*n_jobs"
+    )
+    bench_parameters = dict(
+        output_data_size=int(1e5)  # output data size in bytes,
+    )
+
+    def setup(self, input_size, n_jobs):
+        # Each benchmark is determined by its parameter set and the
+        # duration profile of the tasks it has to execute. The following lines
+        # defines a variety of task duration profiles.
+        random_state = np.random.RandomState(42)
+        high_variance = np.abs(
+            random_state.normal(loc=0.000001, scale=0.01, size=5000)
```

Comment:
For `float` like this, I'd rather use `1e-6` as it is easier to read in my opinion